cider/exec/template/ColumnIR.cpp

```diff
@@ -228,6 +229,28 @@ std::unique_ptr<CodegenColValues> CodeGenerator::codegenFixedLengthColVar(
   return std::make_unique<FixedSizeColValues>(dec_val_cast, null);
 }
 
+std::unique_ptr<CodegenColValues> CodeGenerator::codegenVarCharColVar(
+    const Analyzer::ColumnVar* col_var,
+    llvm::Value* col_byte_stream,
+    llvm::Value* pos_arg,
+    const CompilationOptions& co) {
+  AUTOMATIC_IR_METADATA(cgen_state_);
+  const size_t size = 8;
+  VarcharDecoder decoder(
+      size, &cgen_state_->ir_builder_, !col_var->get_type_info().get_notnull());
+  std::vector<llvm::Instruction*> values =
+      decoder.codegenDecode(cgen_state_->module_, col_byte_stream, pos_arg);
+  for (auto v : values) {
+    cgen_state_->ir_builder_.Insert(v);
+  }
+  llvm::Instruction* null = nullptr;
+  if (values.size() == 3) {
+    null = values[2];
+    values.pop_back();
+  }
```

Comment:
can we assume `null` is always the last one of `values`?

Reply:
`VarcharDecoder` will return 2 or 3 values depends on whether nullable.

Reply:
Actually, ```VarChar``` type could be assumed has three ```Value*``` members but ```Value*``` of null may be ```nullptr```, just as ```FixedSizeColValue``` did. Btw, interfaces of ```CodegenColValue``` should clearly indicate semantics of the underlying ```Value*``` in my opinion.